eventsourcing/src/main/java/org/axonframework/eventsourcing/EntityEvolvingComponent.java

```diff
@@ -17,27 +17,25 @@
 package org.axonframework.eventsourcing;
 
 import org.axonframework.eventhandling.EventMessage;
-import org.axonframework.messaging.unitofwork.ProcessingContext;
+import org.axonframework.messaging.QualifiedName;
 
-import javax.annotation.Nonnull;
+import java.util.Set;
 
 /**
- * Functional interface describing state changes made on a model of type {@code M} based on a given
- * {@link EventMessage}.
+ * Interface describing a group of {@link EntityEvolver EntityEvolvers} belonging to a single entity of type {@code E}.
+ * <p>
+ * The {@link #supportedEvents()} describes the events supported by this entity evolver.
  *
- * @param <M> The model to apply the event state to.
+ * @param <E> The entity type to evolve.
  * @author Steven van Beelen
  * @since 5.0.0
  */
-@FunctionalInterface
-public interface EventStateApplier<M> {
+public interface EntityEvolvingComponent<E> extends EntityEvolver<E> {
```

Comment:
`EntityEvolvingComponent` - sounds weird, I would expect `EventSourcedComponent`, probably. But we may leave it for now. We will see in the future if it's only for entity or also maybe for projections.

Reply:
Hmm, I actually like calling it the `EventSourcedComponent`...I'll simply make that rename and merge it. We'll see how we like it in the months to come!